### PR TITLE
Define HAVE_ARCHIVE to allow building without libarchive

### PR DESCRIPTION
**Pull request by drott**
Redefine std::unique_ptr helper Unique_Archive based on forward
declaration, define HAVE_ARCHIVE in CMakeLists.txt and use it
for conditional compilation of tar related setters.

### CODE REVIEW

File: fuzzing/src/targets/CMakeLists.txt [hunk 1]
```diff
@@ -88,7 +88,7 @@ add_library(fuzztargets
 
 target_compile_definitions(fuzztargets
   PRIVATE
-  "${LOGGER_NAME}")
+  "${LOGGER_NAME}" HAVE_ARCHIVE)
```
Comment (cherusker) [inline fuzzing/src/targets/CMakeLists.txt:91]: if you don't mind let's put `HAVE_ARCHIVE` into a new line; it's a bit of a guess but FreeType's code is usually quite vertically structured

**Review by cherusker — COMMENTED** [review]
very nice to see some movement in this repo after all this time 😊 btw feel free to tag me as a reviewer in all your future (?) PRs -- I should get notifications on my phone then too + can respond quicker 😊

**Comment by drott** [comment]
Thanks for the review, I'll update the CL. I was trying to tag reviewers, but I don't think I have permissions on this repo. Can you make me a contributor?

**Comment by cherusker** [comment]
@drott yeah I was just thinking the same; usually I'm not managing my GH repos (got used to having devops for that); sorry for the inconvenience 😉 and -- very happy for any suggestions of course to make it nice + easy to contribute 😊

File: fuzzing/src/iterators/faceloaditerator.cpp [hunk 2]
```diff
@@ -46,14 +46,15 @@
     (void) face_loader->set_raw_bytes( data, size );
   }
 
-
+#ifdef HAVE_ARCHIVE
```
Comment (cherusker) [inline fuzzing/src/iterators/faceloaditerator.cpp:49]: ➕

File: fuzzing/src/iterators/faceloaditerator.h [hunk 3]
```diff
@@ -61,11 +61,12 @@ namespace freetype {
                    size_t          size );
 
 
+#ifdef HAVE_ARCHIVE
     // @See: FaceLoader::set_data_is_tar_archive()
 
     void
     set_data_is_tar_archive( bool  is_tar_archive );
-    
+#endif
```
Comment (cherusker) [inline fuzzing/src/iterators/faceloaditerator.h:69]: ➕

File: fuzzing/src/targets/FaceFuzzTarget.h [hunk 4]
```diff
@@ -67,11 +67,12 @@ namespace freetype {
     set_supported_font_format( FaceLoader::FontFormat  format );
 
 
+#ifdef HAVE_ARCHIVE
     // @See: `FaceLoader::set_data_is_tar_archive()'
 
     void
     set_data_is_tar_archive( bool  is_tar_archive );
-
+#endif
```
Comment (cherusker) [inline fuzzing/src/targets/FaceFuzzTarget.h:75]: ➕

File: fuzzing/src/utils/faceloader.cpp [hunk 5]
```diff
@@ -60,22 +60,25 @@
...
     else
+#endif
       (void) files.emplace_back( data, data + size );
 
     num_faces     = -1;
     num_instances = -1;
   }
 
-
+#ifdef HAVE_ARCHIVE
```
Comment (cherusker) [inline fuzzing/src/utils/faceloader.cpp:74]: ➕

File: fuzzing/src/utils/utils.h [hunk 6]
```diff
@@ -28,18 +28,18 @@
...
 
-#include <archive.h>
-
 #include <ft2build.h>
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
 
 
+struct archive;
+
```
Comment (cherusker) [inline fuzzing/src/utils/utils.h:37]: not sure exactly about this case but let's maybe also add a double empty line after the declaration

**Review by cherusker — COMMENTED** [review]
lovely thanks; last thing I'd ask for would be to keep the double empty lines. They might look unconventional + accidental but are quite consistenly part of FT's C(++) code style. After that I'm very happy to merge this! 😊

File: fuzzing/src/iterators/faceloaditerator.cpp (hunk 2 above)
Comment (drott) [reply fuzzing/src/iterators/faceloaditerator.cpp]: Done.

File: fuzzing/src/targets/FaceFuzzTarget.h (hunk 4 above)
Comment (drott) [reply fuzzing/src/targets/FaceFuzzTarget.h]: Done.

File: fuzzing/src/utils/utils.h (hunk 6 above)
Comment (drott) [reply fuzzing/src/utils/utils.h:37]: Done.

File: fuzzing/src/utils/faceloader.cpp (hunk 5 above)
Comment (drott) [reply fuzzing/src/utils/faceloader.cpp:74]: Done.

**Review by cherusker — APPROVED** [review]
Thanks so much! 😊

**Pull request merged**